### PR TITLE
feat(models): add report filename option

### LINKED ISSUE #161: Add optional `filename` to `PersistConfig`
`--filename=report` // report.json, report.md

Resources:
- [lighthouse](https://github.com/GoogleChrome/lighthouse)
- [lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci)

### PR DESCRIPTION
**Pull request by BioPhoton**
The changes should help to improve testing of report generation. ATM all tests write in the same folder and with the same file name. Adding a `filename` option to core would improve the testing situation and is helpful in for integrations.

This PR includes:
- extending `persistConfigSchema` with `filename` and a default value
- add tests
- adopt models fixtures to follow new standard see #173 

Implementing the CLI option is resolved in a followup PR

closes #161

### CODE REVIEW

File: packages/models/src/lib/implementation/utils.spec.ts [hunk 1]
```diff
@@ -32,6 +37,32 @@ describe('slugRegex', () => {
...
+    'file:name',
+    'file*name',
+    'file?name',
+    'file"name',
+    'file<name',
+    'file>name',
+    'file|name',
+  ])(`should not match invalid slugs %p`, invalidSlugs => {
+    expect(invalidSlugs).not.toMatch(filenameRegex);
+  });
```
Comment (matejchalk) [inline packages/models/src/lib/implementation/utils.spec.ts:63]:
```suggestion
  ])(`should not match invalid file name %p`, invalidFilename => {
    expect(invalidFilename).not.toMatch(filenameRegex);
  });
```

File: packages/models/src/lib/implementation/utils.spec.ts [hunk 2]
```diff
@@ -32,6 +37,32 @@ describe('slugRegex', () => {
   });
 });
 
+describe('filenameRegex', () => {
+  // test valid and array of strings against filenameRegex with it blocks
+  it.each(['report', 'report.mock', 'report-test.mock'])(
+    `should match valid %p`,
+    slug => {
+      expect(slug).toMatch(filenameRegex);
+    },
```
Comment (matejchalk) [inline packages/models/src/lib/implementation/utils.spec.ts:46]:
```suggestion
    filename => {
      expect(filename).toMatch(filenameRegex);
    },
```

File: packages/models/src/lib/implementation/schemas.ts [hunk 3]
```diff
@@ -102,6 +102,20 @@ export function filePathSchema(description: string) {
...
+/**
+ * Schema for a fileNameSchema
+ * @param description
+ */
+export function fileNameSchema(description: string) {
+  return z
+    .string({ description })
+    .trim()
+    .regex(filenameRegex, {
+      message: 'The filename cant include / : * ? " < > |',
```
Comment (matejchalk) [inline packages/models/src/lib/implementation/schemas.ts:114]:
```suggestion
      message: "The filename can't include / : * ? " < > |",
```

File: e2e/cli-e2e/tests/print-config.spec.ts [hunk 4]
```diff
@@ -40,7 +40,10 @@ describe('print-config', () => {
         apiKey: 'e2e-api-key',
         server: 'https://e2e.com/api',
       },
-      persist: { outputDir: `tmp/${ext}` },
+      persist: {
+        outputDir: `tmp/${ext}`,
```
Comment (matejchalk) [inline e2e/cli-e2e/tests/print-config.spec.ts:44]:
```suggestion
        outputDir: join('tmp', 'ext'),
```

File: e2e/cli-e2e/tests/print-config.spec.ts [hunk 5]
```diff
@@ -64,6 +67,7 @@ describe('print-config', () => {
       },
       persist: {
         outputDir: `tmp/ts`,
```
Comment (matejchalk) [inline e2e/cli-e2e/tests/print-config.spec.ts:69]:
```suggestion
        outputDir: join('tmp', 'ts'),
```

File: packages/models/test/fixtures/report.mock.ts [hunk 6]
```diff
@@ -6,7 +6,7 @@ import { lighthousePluginReport } from './lighthouse-plugin.mock';
 export function report(): Report {
   return {
     packageName: '@code-pushup/core',
-    version: '0.1.0',
+    version: '0.0.1',
```
Comment (matejchalk) [inline packages/models/test/fixtures/report.mock.ts:9]: See https://github.com/flowup/quality-metrics-cli/pull/139#discussion_r1375427865

**Review by matejchalk — CHANGES_REQUESTED** [review]

File: packages/models/test/fixtures/report.mock.ts (hunk 6 above)
Comment (BioPhoton) [reply packages/models/test/fixtures/report.mock.ts:9]: should be addressed in a followup PR

**Review by matejchalk — COMMENTED** [review]
It works nicely with the config :+1: 

The PR description is a bit misleading, `--filename` (or `--persist.filename`) doesn't work as a CLI argument. You'd have to add `persist.filename` to [core-config-options.ts](https://github.com/flowup/quality-metrics-cli/blob/main/packages/cli/src/lib/implementation/core-config-options.ts) to make it accessible from the CLI (we should auto-sync those in the future, probably).

Also, I noticed the PR title doesn't follow our commit conventions - `model` scope should be `models`.

**Review by matejchalk — APPROVED** [review]

**Review by vmasek — DISMISSED** [review]

**Review by matejchalk — APPROVED** [review]

**Pull request merged**